scraping/spiders/mapahelp.py

```diff
@@ -0,0 +1,101 @@
+import re
+import logging
+
+import scrapy
+
+from config.constants import DEFAULT_CATEGORY
+
+
+log = logging.getLogger(__name__)
+
+
+SOCIAL_PATTERNS = [
+    ("messenger", re.compile(r"(https://www\.facebook\.com/[^,;\s]+)")),
+    ("telegram", re.compile(r"(https://t\.me/[^,;\s]+)")),
+    ("telegram", re.compile(
+        r"(?:telegram|телеграм)[-:\s]*@?([_+a-z0-9]+)", flags=re.I)),
+    ("telegram", re.compile(
+        r"@?([_+a-z0-9]+)[-\s]*(?:\(telegram|телеграм\))", flags=re.I)),
+    ("whatsapp", re.compile(
+        r"(?:whatsapp|вотсап|ватсап|вацап)[-:\s]*@?([-_+a-z0-9]+)",
+        flags=re.I)),
+    ("whatsapp", re.compile(
+        r"@?([-_+a-z0-9]+)[-\s]*(?:\(whatsapp|вотсап|ватсап|вацап\))",
+        flags=re.I)),
+]
+
+
+CATEGORIES = {
+    "Finance": "Finance",
+    "Food": "Food",
+    "Health": "Medical",
+    "Housing": "Accommodation",
+    "Humanitarian Aid": "Any help",
+    "Kennels": "Pets",
+    "Psychological Help": "Mental help",
+    "Shelter": "Accommodation",
+    "Sports": "Any help",
+    "Transfer": "Transport",
+    "Other Help": "Any help",
+}
+
+
+def extract_social(txt):
```

Comment:
Is this logic reflected in any of the unit tests?

Reply:
Nope, haven't written any tests for it. I'll write some if it's desirable. So far I observed it extracting most of the data correctly, only weird cases like `+12345678 (all viber, whatsapp, telegram)` and other irregular formats were omitted — people are very creative sometimes.

Reply:
Ok, it's up to you